lib/util/utils.js

```diff
@@ -758,3 +758,43 @@ function isPropertyRequired(propName, model) {
  * Contains the reverse mapping of http.STATUS_CODES
  */
 exports.statusCodeStringToStatusCode = lodash.invert(lodash.mapValues(http.STATUS_CODES, value => { return value.replace(/ |-/g, "").toLowerCase(); }));
+
+
+/**
+ * Models an ARM cloud error.
+ */
+exports.CloudError = {
```

Comment:
```javascript
exports.CloudErrorSchema = {
  description: "Error response describing why the operation failed.",
  schema: {
     "$ref": "#/definitions/CloudErrorWrapper"
   }
};

export.CloudErrorWrapper = {
  properties: {
      error: {
         "$ref": "#/definitions/CloudError"
      }
   },
  additionalProperties: false
};

exports.CloundError = {
  properties: {
     code: {
        type: "string",
        description: "An identifier for the error. Codes are invariant and are intended to be consumed programmatically."
     },
      message: {
            type: "string",
            description: "A message describing the error, intended to be suitable for display in a user interface."
       },
       target: {
            type: "string",
            description: "The target of the particular error. For example, the name of the property in error."
       },
       details: {
            type: "array",
            items: { type: "object" },
            description: "A list of additional details about the error."
       },
       additionalInfo: {
            type: "array",
            items: { type: "object" },
            description: "A list of additional info about an error."
       }
    },
     additionalProperties: false
  };
```

Reply:
This should keep things in sync with our current design.  Do not forget `"type": "object"` in the model definitions for CloudErrorWrapper and CloudError. Now you know the consequences of missing `"type"` on a model definitions. Couple of questions:
1. CloudErrorWrapper has only one property "error". Shouldn't that be a required property?
2. Are there any required properties in `"CloudError"` model? I think at minimum there should be code and message. Otherwise what is the point of the error object.

Reply:
https://github.com/Azure/azure-resource-manager-rpc/blob/master/v1.0/common-api-details.md#error-response-content - code and message are required.

Reply:
thanks for the link, I was looking in the specs repo and those were not marked as required. Btw should we live the innerError and add the new additionalInfo ? (If you saw the email from Gaurav)

Reply:
Yes, I saw that email. I think we should have CloudErrorV1 and CloudErrorV2. Looks like some teams will be providing innerErrors for the already deployed api-versions. We can make our CloudError object a oneOf [V1, V2]. If we do that then we should make sure to not add CloudError during semantic validation, since oneOf is not supported in 2.0 version of the swagger spec.

Reply:
well are we sure those are required  ?either way this change is not intended to catch invalid cloud errors, is mostly to avoid noise from those and potentially catch any other non-described errors. So maybe I would just leave both innerError and additionalInfo. We can refine in the future